cli/agreementbot/getcache.go

```diff
@@ -0,0 +1,216 @@
+package agreementbot
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/open-horizon/anax/agreementbot"
+	"github.com/open-horizon/anax/cli/cliutils"
+	"github.com/open-horizon/anax/i18n"
+	"os"
+)
+
+// Display served pattern orgs and deployment policy orgs cached by aggrement bot.
+func GetServedOrgs() {
+	// set env to call agbot url
+	if err := os.Setenv("HORIZON_URL", cliutils.GetAgbotUrlBase()); err != nil {
+		cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, i18n.GetMessagePrinter().Sprintf("unable to set env var 'HORIZON_URL', error %v", err))
+	}
+
+	// Get the agbot servedorgs info
+	servedOrgsInfo := agreementbot.ServedOrgs{} // the structure we will output
+	cliutils.HorizonGet("cache/servedorg", []int{200}, &servedOrgsInfo, false)
+
+	// Output the combined info
+	jsonBytes, err := json.MarshalIndent(servedOrgsInfo, "", cliutils.JSON_INDENT)
+	if err != nil {
+		cliutils.Fatal(cliutils.JSON_PARSING_ERROR, i18n.GetMessagePrinter().Sprintf("failed to marshal 'hzn node list' output: %v", err))
+	}
+	fmt.Printf("%s\n", jsonBytes)
+
+}
+
+// Display patterns cached by agreement bot. If no org or org/name is specified, display all. Show detailed info if long
+func GetPatterns(org string, name string, long bool) {
+	// set env to call agbot url
+	if err := os.Setenv("HORIZON_URL", cliutils.GetAgbotUrlBase()); err != nil {
+		cliutils.Fatal(cliutils.CLI_GENERAL_ERROR, i18n.GetMessagePrinter().Sprintf("unable to set env var 'HORIZON_URL', error %v", err))
+	}
+
+	// base url upon which to add arguments and flag
+	patUrl := "cache/pattern"
+	// if org is specified add it to url
+	// if name is specified add it to url and ignore value for long
+	if org != "" {
+		patUrl = fmt.Sprintf("%v/%v", patUrl, org)
+		if name != "" {
+			patUrl = fmt.Sprintf("%v/%v", patUrl, name)
+			long = false
+		}
+	}
+
+	if long {
+		// Show detailed info if long, else only orgs and names
+		if org == "" {
+			// Get the agbot servedorgs info
+			patInfo := map[string]map[string]*agreementbot.PatternEntry{} //the structure we will output
```

Comment:
You forgot to call the agbot API here

Reply:
Done. Also rearranged some variable assignments to reduce redundancy.